### PR TITLE
MlflowLogger limit parameter value length 250 char

### LINKED ISSUE #5892: MlflowLogger fail when logging long parameters
### To Reproduce

Log anything  parameters longer than 250 characters

### Expected behavior

MlflowLogger not sending parameters longer than 250 characters to mlflow and log warning to user

### Environment

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux): 
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

### Additional context

Mlflow only allow paramters to be at most 500 bytes (250 unicode characters), their limit in database is 250 characters:
https://www.mlflow.org/docs/latest/rest-api.html#log-param
https://github.com/mlflow/mlflow/issues/1976
https://github.com/mlflow/mlflow/issues/3931

### PR DESCRIPTION
**Pull request by ducthienbui97**
## What does this PR do?

Fixes #5892 (issue) <- this [links related issue to this PR](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@ducthienbui97 pls rebase on feat 1.2 see
https://pytorch-lightning.readthedocs.io/en/latest/CONTRIBUTING.html#how-to-fix-pr-with-mixed-base-and-target-branches

**Comment by ducthienbui97** [comment]
@Borda Thanks, updated

File: pytorch_lightning/loggers/mlflow.py [hunk 1]
```diff
@@ -150,6 +150,10 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
         params = self._convert_params(params)
         params = self._flatten_dict(params)
         for k, v in params.items():
+            if len(str(v)) > 250:
+                log.warning(f"Mlflow only allows parameters with up to 250 characters. Discard {k}={v}")
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply pytorch_lightning/loggers/mlflow.py:154]: also let's use `rank_zero_warn` here instead of log.warning?

**Review by awaelchli — COMMENTED** [review]
would be nice to cover this line. Have a look at `tests/loggers/test_mlflow.py`
You can use (pseudo code):
```python
with pytest.warns(UserWarning, match="Discard x = y")
    mlflow.log_ ...
```
ping me if you need help :)

File: pytorch_lightning/loggers/mlflow.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ducthienbui97) [reply pytorch_lightning/loggers/mlflow.py:154]: @awaelchli yeah, it's v, there are feature requests in mlflow about that
https://github.com/mlflow/mlflow/issues/3931
https://github.com/mlflow/mlflow/issues/1976

**Review by awaelchli — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by Borda — APPROVED** [review]

**Pull request merged**